### PR TITLE
Toponaming make shape with element map

### PR DESCRIPTION
**Pull request by chennes**
This is a composite PR from real thunder's original code, plus contributions from me, @bgbsww, and @CalligaroV. It adds `makeShapeWithElementMap` (original name :`makESHAPE`) as well as a set of tests for it and its dependent methods.

### CODE REVIEW

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 1]
```diff
@@ -509,6 +514,743 @@ void TopoShape::mapSubElement(const std::vector<TopoShape>& shapes, const char*
...
+
+    int find(const TopoDS_Shape& subshape)
+    {
+        return cache.find(shape, subshape);
+    }
+};
+
+////////////////////////////////////////
+// makESHAPE -> makeShapeWithElementMap
+///////////////////////////////////////
```
Comment (bgbsww) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:549]: Is this ready to be removed?

File: tests/src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 2]
```diff
@@ -125,30 +123,10 @@ TEST_F(TopoShapeExpansionTest, makeElementCompoundTwoShapesGeneratesMap)
...
-
-    return {box1, box2};
-}
-}  // namespace
-
 TEST_F(TopoShapeExpansionTest, makeElementCompoundTwoCubes)
 {
     // Arrange
-    auto [cube1, cube2] = CreateTwoCubes();
+    auto [cube1, cube2] = PartTestHelpers::CreateTwoCubes();
```
Comment (bgbsww) [inline tests/src/Mod/Part/App/TopoShapeExpansion.cpp:129]: using namespace PartTestHelpers;

Maybe?  not critical.

**Review by bgbsww — COMMENTED** [review]
A couple trivialities.  Looks good.

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 3]
```diff
@@ -509,6 +514,743 @@ void TopoShape::mapSubElement(const std::vector<TopoShape>& shapes, const char*
...
+    static std::string postfix(TopoShape::elementMapPrefix() + ":U");
+    return postfix;
+}
+
+const std::string& lowerPostfix()
+{
+    static std::string postfix(TopoShape::elementMapPrefix() + ":L");
+    return postfix;
+}
+
```
Comment (Pesc0) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:636]: is it the same as this https://github.com/FreeCAD/FreeCAD/blob/main/src/App/ElementNamingUtils.h?

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 1 above)
Comment (chennes) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:549]: I was actually planning on leaving it until we were basically done, as a sort of breadcrumb to search for. I could be dissuaded though.

File: tests/src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 2 above)
Comment (chennes) [reply tests/src/Mod/Part/App/TopoShapeExpansion.cpp:129]: I have no strong feeling, so I'm happy to change it.

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 3 above)
Comment (chennes) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:636]: Yeah, I thought I had gone back and referenced the constants, but apparently not.

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 1 above)
Comment (bgbsww) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:549]: Nope, that's fine!

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 3 above)
Comment (chennes) [reply src/Mod/Part/App/TopoShapeExpansion.cpp]: OK, that's done now.

**Pull request merged**